packages/pangea-sdk/pangea/request.py

```diff
@@ -25,11 +24,7 @@ class PangeaRequest(object):
     """
 
     def __init__(
-        self,
-        config: PangeaConfig,
-        token: str,
-        version: str,
-        service: str,
+        self, config: PangeaConfig, token: str, version: str, service: str, logger: Optional[logging.Logger] = None
```

Comment:
I still don't like this. Literally all I think we need is
```
logger_name: str = "pangea"
```

```
self.logger = logging.getLogger(logger_name)
```

and the user can set up the dictConfig for their logger however they want. If they're using us in their [django](https://docs.djangoproject.com/en/4.1/topics/logging/) application, they can then do (as an example)

```
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
        },
    },
    'root': {
        'handlers': ['console'],
        'level': 'WARNING',
    },
    'loggers': {
        'django': {
            'handlers': ['console'],
            'level': os.getenv('DJANGO_LOG_LEVEL', 'INFO'),
            'propagate': False,
        },
        'pangea': {
            'handlers': ['console'],
            'level': os.getenv('PANGEA_LOG_LEVEL', 'ERROR'),
            'propagate': False,
        },
    },
}
```

the point is, all we do is emit to a named log in python, they can do whatever config they want beyond that.

We do the same thing with common third party libraries:
- For aws packages, we configure logging configs for loggers named "boto3" and "botocore"
- For django, we configure logging for "django", "djagno.security", etc
- For requests, we configure logging for "requests"

We probably don't even need a `LoggerConfig` object

Reply:
Cool then. I'll do this way and add a function on utils to set up our default settings